scripts/check-changelog.py

```diff
@@ -41,11 +42,21 @@
         for d in (now, now + hour, now - hour)
     ))
 
-    for line in tools.changelog().split('\n'):
-        if line.strip() in acceptable_lines:
+    pattern = r'\n\d+\.\d+\.\d+ - \d\d\d\d-\d\d-\d\d\n'
+    all_versions = list(map(str.strip, re.findall(pattern, tools.changelog())))
+
+    for line in acceptable_lines:
+        if line == all_versions[0]:
             break
+        elif line in all_versions:
+            print('Changelog entry %r is not the first entry!  Check for '
+                  'merge errors.' % line)
+            sys.exit(1)
     else:
         print('No line with version and current date (%s) in the changelog. '
               'Remember this will be released as soon as you merge to master!'
               % ' or '.join(repr(line) for line in acceptable_lines))
         sys.exit(1)
+
+    assert all_versions == sorted(all_versions), \
+        'You edited old release notes and changed the ordering!'
```

Comment:
Personally, I’d make this an `if` statement instead:

```python
if all_versions != sorted(all_versions):
```

I reserve `assert` for "the program has got into an unexpected/should-be-impossible state, abort!" – whereas it's totally possible that one of us screws up the changelog. 😉